### PR TITLE
Task14

### PR DESCRIPTION
**Pull request by yana-may25**

### CODE REVIEW

File: task14/task14.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+TextCell.prototype.minWidth = function () {
+    return this.text.reduce(function (width, line) {
+        return Math.max(width, line.length);
+    }, 0);
+};
+
+TextCell.prototype.minHeight = function () {
+    return this.text.length;
+};
+
```
Comment (MrOlegKravchenko) [inline task14/task14.js:30]: Add both _TextCell.prototype.draw_ and _StretchCell.prototype.draw_ functionality from the topic
With console log check for it

Comment (yana-may25) [reply task14/task14.js:30]: Ok, so I added those draw functions, I even think I did that as in the book
BUT it is not **inheritance**, it is **composition**. Like, textCell is a part of stretchCell. And _draw_ method is not redefined, it's a new one (I know I have different parameters in the two draws, but it's not the point)
ooooor I have been inattentive when reading, as usual xD

File: task14/task14.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+StretchCell.prototype = Object.create(TextCell.prototype);
+
+function StretchCell(inner, width, height) {
+    this.inner = inner;
+    this.width = width;
+    this.height = height;
+}
+
+StretchCell.prototype.minWidth = function () { return this.width > this.inner.minWidth() ? this.width : this.inner.minWidth() }
```
Comment (MrOlegKravchenko) [inline task14/task14.js]: Math.max ??

File: task14/task14.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+StretchCell.prototype.minHeight = function () { return this.height > this.inner.minHeight() ? this.height : this.inner.minHeight() }
```
Comment (MrOlegKravchenko) [inline task14/task14.js]: Math.max ??

File: task14/task14.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+StretchCell.prototype.draw = function () {
+    return this.inner.draw(this.minWidth(), this.minHeight())
```
Comment (MrOlegKravchenko) [inline task14/task14.js]: this.inner.draw(width, height) ??

File: task14/task14.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
+function TextCell(text) {
+    this.text = text.split("\n");
+}
+
+TextCell.prototype.draw = function (width, height) {
+    var result = [];
+    for (var i = 0; i < height; i++) {
+        var line = this.text[i] || "";
+        result.push(line + repeat(" ", width - line.length));
```
Comment (MrOlegKravchenko) [inline task14/task14.js:16]: What is **repeat** here?

**Pull request merged**